### PR TITLE
python-wheel: upgrade to 0.43.0

### PR DESCRIPTION
**Pull request by BettyRain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [X] Ready to merge

---

###### Summary 

[What does the PR accomplish, why was it needed?
python-wheel: upgrade to 0.43.0

###### Change Log  

- Upgrade python-wheel to 0.43.0

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local build
- Latest pipeline build id: [569969](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=569969&view=results)
- Delta build id: [567474](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=567474&view=results)
- Full build id: [567492](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=567492&view=results)

### CODE REVIEW

File: SPECS/python-wheel/python-wheel.spec [hunk 1]
```diff
@@ -57,39 +58,50 @@ A Python wheel of wheel to use with virtualenv.
...
 
 %check
-rm setup.cfg
-%{python3} -m pip install pluggy more-itertools
-PYTHONPATH=%{buildroot}%{python3_sitelib} py.test3 -v --ignore build
+# Smoke test
+%{py3_test_envvars} wheel-%{python3_version} version
+%py3_check_import wheel
+
+%if %{with tests}
```
Comment (anphel31) [inline SPECS/python-wheel/python-wheel.spec:98]: should use `with_check` instead

File: SPECS/python-wheel/python-wheel.spec [hunk 2]
```diff
@@ -57,39 +58,50 @@ A Python wheel of wheel to use with virtualenv.
...
 
+mkdir -p %{buildroot}%{python_wheel_dir}
+install -p %{_pyproject_wheeldir}/%{python_wheel_name} -t %{buildroot}%{python_wheel_dir}
+
 
 %check
-rm setup.cfg
-%{python3} -m pip install pluggy more-itertools
-PYTHONPATH=%{buildroot}%{python3_sitelib} py.test3 -v --ignore build
+# Smoke test
```
Comment (anphel31) [inline SPECS/python-wheel/python-wheel.spec:94]: can you move the smoke test inside of the `if with_check` block?

**Comment by BettyRain** [comment]
New runs after PR comments:
Buddy build: [568918](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=568918&view=results)

File: SPECS/python-wheel/python-wheel.spec [hunk 3]
```diff
@@ -57,39 +58,49 @@ A Python wheel of wheel to use with virtualenv.
...
 %check
-rm setup.cfg
-%{python3} -m pip install pluggy more-itertools
-PYTHONPATH=%{buildroot}%{python3_sitelib} py.test3 -v --ignore build
+%if 0%{?with_check}
+# Smoke test
+%{py3_test_envvars} wheel-%{python3_version} version
+%py3_check_import wheel
+%pytest -v --ignore build
+%endif
```
Comment (PawelWMS) [inline SPECS/python-wheel/python-wheel.spec:99]: Redundant - the `%check` section is only run during test runs and that's when `with_check` is always set to `1`, so it should be fine to drop it.
```suggestion
# Smoke test
%{py3_test_envvars} wheel-%{python3_version} version
%py3_check_import wheel
%pytest -v --ignore build
```

File: SPECS/python-wheel/python-wheel.spec [hunk 4]
```diff
@@ -57,39 +58,49 @@ A Python wheel of wheel to use with virtualenv.
...
 
 %files -n python3-%{pypi_name}
 %license LICENSE.txt
 %doc README.rst
-%{_bindir}/%{pypi_name}
-%{_bindir}/%{pypi_name}-3
```
Comment (PawelWMS) [inline SPECS/python-wheel/python-wheel.spec:92]: Are these two supposed to disappear? Considering the names, they sound like the main component for the package.

File: SPECS/python-wheel/python-wheel.spec (hunk 3 above)
Comment (BettyRain) [reply SPECS/python-wheel/python-wheel.spec]: Don't we want to keep at least:
```
%{py3_test_envvars} wheel-%{python3_version} version
%py3_check_import wheel
```
 and remove what they had in
` %if %{with tests} `
section?

File: SPECS/python-wheel/python-wheel.spec (hunk 4 above)
Comment (BettyRain) [reply SPECS/python-wheel/python-wheel.spec:92]: That's fair, I added it the way they add it in the current version.

File: SPECS/python-wheel/python-wheel.spec (hunk 3 above)
Comment (BettyRain) [reply SPECS/python-wheel/python-wheel.spec]: Discussed in DM

**Review by PawelWMS — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Pull request merged**